src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala

```diff
@@ -322,6 +323,7 @@ class GoogleExtensions(
               // See: https://broadworkbench.atlassian.net/browse/CA-1005
               googleDirectoryDAO.addServiceAccountToGroup(proxyEmail, sa)
             }))
+            _ <- IO.fromFuture(IO(googleIamDAO.addIamRoles(project, sa.email, MemberType.ServiceAccount, Set("roles/iam.serviceAccountUser"), retryIfGroupDoesNotExist = true)))
```

Comment:
Posted in slack, but I'm wondering if `addIamRoles` is the right method. I think this will give the pet SA `Service Account User` at the _project_ level, which lets the pet SA impersonate any SA in the project (which is not what we want).

[This method](https://github.com/broadinstitute/workbench-libs/blob/develop/google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleIamDAO.scala#L293) to me seems like what we want -- it applies the `Service Account User` role to a _SA_ with a given target (which happens to be the same SA in this case).

I'm curious if the latter approach worked with NextFlow?

Reply:
oh that makes sense why i've been seeing the 403! Yes I want to add the `Service Account User` role to only the SA not at the project level. I'll update and test out the `addServiceAccountUserRoleForUser` method!

Reply:
I had originally used `addServiceAccountUserRoleForUser` and it didn't work but i didn't look at the sam logs. testing it out again while looking at sam logs to see what's going on. waiting on fiab-start